src/api/app/policies/status/report_policy.rb

```diff
@@ -0,0 +1,53 @@
+class Status::ReportPolicy < ApplicationPolicy
+  def initialize(user, record)
+    raise Pundit::NotAuthorizedError, 'record does not exist' unless record
+    @user = user
+    @record = record
+  end
+
+  def create?
+    return false if @user.blank?
+    return true if @user.is_admin?
+
+    checkable_containers = case @record.checkable
+                           when BsRequest
+                             @record.checkable.bs_request_actions.map { |action| request_target_object(action) }
+                           when Repository
+                             [@record.checkable.project]
+                           end
+
+    checkable_containers.all? { |container| write_permission_for_checkable?(container) }
+  end
+
+  def update?
+    create?
+  end
+
+  def destroy?
+    create?
+  end
+
+  def index?
+    true
+  end
+
+  def show?
+    index?
+  end
+
+  private
+
+  def write_permission_for_checkable?(container)
+    container &&
```

Comment:
I would do `return false unless container` and then the `||`. Much easier to read and you don't have the problem of splitting the lines (at least for the first one)